### PR TITLE
[FLINK-6867] [elasticsearch] Fix instable Elasticsearch 1.x ITCases

### PR DESCRIPTION
**Pull request by tzulitai**
The root cause for the instability in ES 1.x is that we're testing both embedded client mode and transport client mode against the same Elasticsearch environment. This PR ignores the embedded mode testing, as that method is generally discouraged anyways.

### CODE REVIEW

File: flink-connectors/flink-connector-elasticsearch/src/test/java/org/apache/flink/streaming/connectors/elasticsearch/ElasticsearchSinkITCase.java [hunk 1]
```diff
@@ -67,7 +68,11 @@ public void testTransportClientFails() throws Exception{
 
 	/**
 	 * Tests that the Elasticsearch sink works properly using an embedded node to connect to Elasticsearch.
+	 *
+	 * NOTE: This is ignored, since executing both embedded client mode and transport client mode introduces
+	 * instability for node discovery in ES 1.x. See FLINK-6867.
 	 */
+	@Ignore
```
Comment (zentol) [inline flink-connectors/flink-connector-elasticsearch/src/test/java/org/apache/flink/streaming/connectors/elasticsearch/ElasticsearchSinkITCase.java]: Do we expect to ever enable this test again? if not, let's just remove it.

Comment (zentol) [inline flink-connectors/flink-connector-elasticsearch/src/test/java/org/apache/flink/streaming/connectors/elasticsearch/ElasticsearchSinkITCase.java]: that would also resolve the checkstyle violation :P

Comment (tzulitai) [inline flink-connectors/flink-connector-elasticsearch/src/test/java/org/apache/flink/streaming/connectors/elasticsearch/ElasticsearchSinkITCase.java]: I would not expect it to be enabled again, at least for the Travis runs.
At the rate of the new Elasticsearch versions (there's already a 6.x now), we might also consider dropping 1.x support soon.

Makes sense, will remove it completely :)

**Comment by tzulitai** [comment]
Merging once Travis gives green.

**Pull request closed without merging**